lib/services/nodes-api-service.js

```diff
@@ -0,0 +1,96 @@
+// Copyright 2015, EMC, Inc.
+
+'use strict';
+
+var di = require('di');
+
+module.exports = nodeApiServiceFactory;
+di.annotate(nodeApiServiceFactory, new di.Provide('Http.Services.Api.Nodes'));
+di.annotate(nodeApiServiceFactory,
+    new di.Inject(
+        'Protocol.TaskGraphRunner',
+        'Services.Waterline',
+        'Errors',
+        'Logger',
+        '_',
+        'Promise'
+    )
+);
+function nodeApiServiceFactory(
+    taskGraphProtocol,
+    waterline,
+    Errors,
+    Logger,
+    _,
+    Promise
+) {
+    var logger = Logger.initialize(nodeApiServiceFactory);
+
+    function NodeApiService() {
```

Comment:
@anhou, the "Protocol.TaskGraphRunner" is just in on-core, so I think it's proper. Image future we implements the auto node removal, I think this node-api-service is exactly what we need for auto node removal.

Anyway, it's OK for me that move it to on-core after we come across that situation.